### PR TITLE
UI launches crawl

### PR DESCRIPTION
**Pull request by colleenlemak11**
Update to PR https://github.com/SD-2023-CS10/Project-Version-0/pull/55:

Implements:

POST request
Changes to how username is received (NOT via config file)
Changes to related integration documentation
Keeps:

HTML requests
proc_open() built-in method
Piping process in runScan.py (to receive console messages)
Notes:
I tried switching to the simpler logic wrapping the scan button in form tag, but I wasn't able to print any echo/print messages from the files (confirming they are actually running). Optimizing this code could be a post-demo task if we really want to get rid of console messages and/or the output-piping process.

Expected Sequence:
Button click -> runScan.php -> dependency-checks.php and crawl-device.py are launched.

### CODE REVIEW

File: crawler/README.md [hunk 1]
```diff
@@ -47,12 +59,25 @@ Next, information will be displayed for the Target Hostname, Server Gateway IP,
 
 As the script progresses, it will output feedback messages to the terminal indicating the ongoing operations.  Note that port-scanning typically consumes the majority of the scan time.
 
+## Integration Process
+Integration of the crawler Python file with other components posed several challenges. The crawler solution consists of helper files to download dependencies, pass information from file-to-file, and pipe feedback messages to the terminal in order to confirm integrity of important data (i.e. the username of the client currently logged in).
+
+1. Upon the repository tool's bootup, `login.html` prompts the user for their credentials.  This information is dynamically written to the UI's config.ini file for parsing and storage.
```
Comment (BrandonHuyck) [inline crawler/README.md:65]: I'm not sure this is true. The login page should be for the user/pass for our app. The config file is for db credentials--these are not getting set dynamically in our current version ("Beyond scope").

File: UI/runScan.php [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    using: Fn F12.
+ -->
+<?php
+
+    // Allow capturing output buffer
+    ob_implicit_flush(true);
+    
+    // Locate and parse configuration file
+    // $config = parse_ini_file("./config.ini");
+    // $username = $config["username"];
```
Comment (BrandonHuyck) [inline UI/runScan.php:14]: The username for our app is different from the username for the database. The database credentials (user/pass) are fixed; any sort of dynamic validation/setting is beyond our scope. The username that we are passing around through code and that the user inputs on the login screen is solely their user/pass for our app, which tells the DB what client they are associated with.

File: UI/runScan.php [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    Note: output messages built-into the crawler script are not visible on the server-side while running localhost:<port#> 
+    (i.e. during demonstration). However, at the end of the scan, output will be logged to the UI console available to view 
+    using: Fn F12.
+ -->
+<?php
+
+    // Allow capturing output buffer
+    ob_implicit_flush(true);
+    
+    // Locate and parse configuration file
```
Comment (BrandonHuyck) [inline UI/runScan.php:12]: The config file only needs to be parsed when you are manually connecting to the database

File: UI/runScan.php (hunk 2 above)
Comment (BrandonHuyck) [inline UI/runScan.php:14]: The username will be stored in a PHP `$_SESSION[<usr>]` variable. See: [link](https://www.w3schools.com/php/php_sessions.asp).

File: crawler/printhi.py
Comment (BrandonHuyck) [inline crawler/printhi.py:1]: The purpose of this file is unclear. Also, the name should probably be changed, unless that is clarified with its purpose.

File: UI/runScan.php [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+            }
+        } else {
+            echo "Error: PHP script not found.";
+        } 
+    }
+
+    // Launches crawler/crawl-device.py with username as argument from Project-Version-0 folder
+    function runScan($username) {
+        // Navigate from /UI, up to parent dir then to crawler file
+        // $crawlerScriptRelativePath = '/../crawler/printhi.py';
```
Comment (BrandonHuyck) [inline UI/runScan.php:66]: Lines 66-67 are unclear to me.

**Review by BrandonHuyck — CHANGES_REQUESTED** [review]
See comments. Notable, 1) there seems to be some misunderstanding around the difference between database and application credentials; and 2) there needs to be more clarity with `printhi`.

**Comment by colleenlemak11** [comment]
Update to PR #55: 

Implements:
- POST request
- Changes to how username is received (NOT via config file)
- Changes to related integration documentation

Keeps:
- HTML requests
- proc_open() built-in method
- Piping process in runScan.py (to receive console messages)  

Notes:
I tried switching to the simpler logic wrapping the scan button in <form>, but I wasn't able to print any echo/print messages from the files (confirming they are actually running). Optimizing this code could be a post-demo task if we really want to get rid of console messages and/or the output-piping process.

Expected Sequence:
Button click -> runScan.php -> dependency-checks.php and crawl-device.py are launched.

**Review by BrandonHuyck — APPROVED** [review]

**Pull request merged**